Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -298,15 +300,18 @@ private extension UInt8 {
         case .keepAlive:
             return true
         case .unknown:
-            guard let connection = self["connection"].first?.lowercased() else {
-                // HTTP 1.1 use keep-alive by default if not otherwise told.
-                return version.major == 1 && version.minor == 1
-            }
-            
-            if connection == "close" {
-                return false
+            if !self.headers.isEmpty {
+                for header in self.headers {
+                    if self.buffer.equalCaseInsensitiveASCII(view: connectionUtf8, at: header.name) {
+                        if self.buffer.equalCaseInsensitiveASCII(view: closeUtf8, at: header.value) {
```

Comment:
technically this isn't a correct implementation, right? could be `connection: keep-alive, foobar` no?

I know, it's not worse than the previous implementation either but the fast path is anyway that there is no `connection` header

Reply:
@weissi I just preserved the old behaviour. @Lukasa thoughts ?